modules/applications/pages/ui-builder/how-to-localize-your-application.adoc

```diff
@@ -0,0 +1,105 @@
+For example, `This is a sample application to demonstrate the possibility of making it available in multiple languages.`
+
+
+=== 2 Define JS objects
+* Go to the JS section, create a new JavaScript object, and name it localization.
+* In the `localization` file, define JavaScript objects for each language to correspond with the values set for the `SelectLanguage` widget. Include the words or sentences along with their translations as key-value pairs for all languages, as shown in the code snippet below:
+
+[source, JS]
+----
+export default {
+	translations: {
+		en: {
+			"This is a sample application to demonstrate the possibility of making it available in multiple languages.": "This is a sample application to demonstrate the posibility of making it available in multi-languages."
+		},
+		fr: {
+			"This is a sample application to demonstrate the possibility of making it available in multiple languages.": "Il s'agit d'un exemple d'application visant à démontrer la possibilité de le rendre disponible en plusieurs langues."
+		},
+		es: {
+			"This is a sample application to demonstrate the possibility of making it available in multiple languages.": "Esta es una aplicación de muestra para demostrar la posibilidad de hacerlo disponible en varios idiomas."
+		}
+	}
+}
+----
+
+
+[NOTE]
+====
+Ensure that all language keys match the values set in the SelectLanguage source data, such as en, fr, and es, etc.
+====
+
+* To keep the codebase organized, define another JavaScript object and name it i18n.
+* Copy and paste the JavaScript code blocks provided below:
+
+[source, JS]
+----
+export default {
+	languageKeys: {},
+	onLanguageChange() {
+		this.languageKeys = localization.translations[SelectLanguage.selectedOptionValue];
+	},
+	setLangugeKey() {
+		this.languageKeys = localization.translations["en"];
+		return this.languageKeys;
+	}
+}
+----
+
+This JavaScript module manages language translations using the `languageKeys` property. The `onLanguageChange()` method updates `languageKeys` based on the selected language, while `setLangugeKey()` sets it to the default English translations and returns the updated object.
+
+[NOTE]
+====
+Ensure that the `Run on page load` setting is enabled for the `setLanguageKey` function.
+====
+
+
+=== 3 Bind JS objects to the widgets
+* Come back to the UI section and select the `SelectLanguage` widget.
+* Set its `Label` to `{{i18n.languageKeys["Select language"]}}`.
+* In the Events property, set `{{i18n.onLanguageChange();}}` to invoke the function when the language changes.
+* Then, select the text widget and set the `Text` property to `{{i18n.languageKeys["This is a sample application to demonstrate the possibility of making it available in multiple languages."]}}`.
+
+
+[NOTE]
+====
+"Since the goal of this example is to provide a workaround, it does not address cookies. You can search online for information and implement cookie management according to your needs."
```

Comment:
```suggestion
Since the goal of this example is to provide best practices on how to manage several languages, it does not address cookies. It is of course possible to define JS objects which manage cookies according to your needs.
```